qiskit/primitives/backend_sampler.py

```diff
@@ -0,0 +1,234 @@
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Sampler implementation for an artibtrary Backend object."""
+from __future__ import annotations
+
+from collections.abc import Iterable, Sequence
+import copy
+from typing import cast, Any
+
+from qiskit.exceptions import QiskitError
+from qiskit.providers.options import Options
+from qiskit.result import QuasiDistribution
+from .base_sampler import BaseSampler
+from .sampler_result import SamplerResult
+from .primitive_job import PrimitiveJob
+
+
+class BackendSampler(BaseSampler):
+    """A :class:`~.BaseSampler` implementation that provides an interface for leveraging
+    the sampler interface from any backend.
+
+    This class provides a sampler interface from any backend and doesn't do
+    any measurement mitigation, it just computes the probability distribution
+    from the counts.
+    """
+
+    def __init__(
+        self,
+        backend: "Backend" = None,
+        circuits: "QuantumCircuit" | Iterable["QuantumCircuit"] = None,
+        parameters: Iterable[Iterable["Parameter"]] | None = None,
+        bound_pass_manager: "PassManager" | None = None,
+        skip_transpilation: bool = False,
+    ):
+        """Initialize a new BackendSampler
+
+        Args:
+            backend: Required: the backend to run the sampler primitive on
+            circuits: The Quantum circuits to be executed.
+            parameters: Parameters of each of the quantum circuits.
+                Defaults to ``[circ.parameters for circ in circuits]``.
+            bound_pass_manager: An optional pass manager to run after
+                parameter binding.
+            skip_transpilation: If this is set to True the internal compilation
+                of the input circuits is skipped and the circuit objects
+                will be directly executed when this objected is called.
+        Raises:
+            ValueError: If backend is not provided
+        """
+
+        super().__init__(circuits, parameters)
+        if backend is None:
+            raise ValueError("A backend is required to use BackendSampler")
+        self._backend = backend
+        self._run_options = self._backend.options
+        self._is_closed = False
+        self._transpile_options = Options()
+        self._bound_pass_manager = bound_pass_manager
+        self._preprocessed_circuits: list["QuantumCircuit"] | None = None
+        self._transpiled_circuits: list["QuantumCircuit"] | None = None
+        self._skip_transpilation = skip_transpilation
+
+    @property
+    def preprocessed_circuits(self) -> list["QuantumCircuit"]:
```

Comment:
This is simply my interest rather than a recommendation to change, but why are you quoting `QuantumCircuit`? Don't you want to import it?

Reply:
I was debugging issues with a cyclical import locally, I assumed it was from the import at one point so I removed a bunch of type hint only imports to try and workaround it. But, it turned out to be a local environment issue and not an import problem. I can change it to use the actual class if you'd prefer